schemas/data.yaml

```diff
@@ -0,0 +1,28 @@
+CREs:
+  - CRE: "<ID in here>" # ID could be a number, or a keyword. If it's a keyword, an ID will be set in the metadata.
```

Comment:
Based on the comment this is imprecise perhaps that's intentional and I'm missing the vision?

Also this is going to sound silly, but if I was a member or dev that had never heard of this standard/effort what does the acronym CRE actually mean? (Common Requirement Enumeration? 🤷‍♂️ )

Reply:
Can you discuss actions for both points? This is still being built, so I need to see what you'd be aiming at precisely.
1. We still didn't pick what should be the identifier for the CRE -- so this is still not set in stone, trying to take feedback on it and we're delving deeper into which to use as the main identifier.
2. Should we set it as Common_Requirement_Enumeration? CWE is used as CWE and everyone uses it as such. ISO, PCIDSS, ASVS; how is this going to be different?

Reply:
We should absolutely discuss that's the whole point of this process right?

1. I'm good with either ID or keyword, however keyword may become difficult as it's more qualitative than quantitative (people will have different definitions/interpretations of keywords). At the same time ID(s) will require more maintenance and perhaps a boolean `isDeprecated` (or whatever) field.
2. I'm good with whatever, but it should be included as a comment somewhere for clarity. Honestly I picked "common requirement enumeration' out of thin air. I dunno what you or others were using terminology wise.

Reply:
Oh, I am saying that CRE is that, and just replacing it with that. I feel like a comment would do to make it more understandable.
I'll have a deeper look into the `ID` on Monday.
Any points on the structure or does it look good for you?

Reply:
@kingthorin
maybe an id could be structured as:
CATEGORY_ID-Num_in_category
so CRE "always encrypt a database"
could be CRYPTO-2
alternatively, if grouping CREs is confusing, it could be just an id
1234-2
1234 being an arbitrary id.

Reply:
The ID naming is something a little bit bigger than this. @northdpole you can see that in the sheet we have. Let's not focus this discussion in here. This PR is not precisely for the ID structure, if you want we can create an issue and get in bigger input.

Reply:
There is no need to distinguish between types of sources (requirements vs tests) here, for two reasons: 1) the type can be ambiguous: even though the testing guide is written as a test guide, it also serves well as a requirement document for developers. Look at the ASVS: it is written as a set of checks, so you could also call that a 'test' source. Better not. And 2) IF we want to distinguish between types of sources, this is better stored once at the metadata itself. That makes this type easier to maintain and the CRE content more elegant.

Reply:
The ID's of the sources (e.g. ASVS-ID) should be optional and not the URI. The Purpose of the CRE is to refer to the right section. Not to keep a list for reference. Having to keep an ID increases maintenance.

Reply:
I would assume the IDs are more stable than URIs. The OWASP Top 10 entry ID `A1:2017` stayed the same when its location moved from <https://owasp.org/index.php/Top_10-2017_A1-Injection> to legacy <https://wiki.owasp.org/index.php/Top_10-2017_A1-Injection> and now redirects to <https://owasp.org/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A1-Injection>. Shouldn't the CREs be more of a "meta list" of well-known requirements than a "link collection"? I'd imagined it like something similar to page 2 of https://www.cisecurity.org/wp-content/uploads/2017/03/Poster_Winter2016_CSCs.pdf just for AppSec.

Reply:
On ID's versus links. There are two reasons for hyperlinks to be the primary pointer to specific content in a standard. 
1) Link-granularity: often you want to link to a specific section in a standard text that does not have its own ID in the standard. For example: in WSTG the part on checking random session ID's is just one of the many paragraphs in the session testing section. You want to point to the right one. In other words: often a CRE topic has no corresponding ID in a source.
2) Self-maintenance: Yes the IDs are more stable and they are very helpful to have in case the hyperlink is outdated - which is why the IDs should be optional at least. At the same time: one of the key design principles of CRE is that it is self-maintaining and therefore that everything has been to make the links work. In the past, every mapping initiative that needed to be maintained by hand  failed. It's a dead end.
In the case of ASVS this should be automatic as the idea is either 1)the CRE-IDs are embedded IN the content of the ASVS or 2)the maintenance process ensures that changes of the CRE-ID mapping are always done when changes in the ASVS take place.
There are other cases where maintenance of CRE in a standard is a separate process and can lack behind. Typically these are standard that are not yet committed to maintaining their CRE links. In that case an ID can be very handy.

Reply:
Regarding CIS controls: yes, that's a good comparison. And there are many more comparisons to make: requirements are controls and guidelines, and rules, and criteria, and checks, and verification items. Basically all standards aim for consensus on these 'things'. The CIS controls are quite high level. CRE aims to be much more towards the level of ASVS. Makes sense?

Reply:
Following some discussions, it was agreed that the CRE ID will follow the digits convention (_e.g._ CRE XXXX). The number of digits and starting point will be discussed in an upcoming call. If you have any insights on this, please let us know as well :smile: